internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/D2ControllerClient.java

```diff
@@ -96,24 +114,27 @@ protected String discoverLeaderController() {
   }
 
   private static <RESPONSE> RESPONSE d2ClientGet(
-      D2Client d2Client,
+      List<D2Client> d2ClientList,
       String d2ServiceName,
       String path,
       QueryParams params,
       Class<RESPONSE> responseClass) {
     String requestPath = D2_SCHEME + d2ServiceName + path + "?" + encodeQueryParams(params);
-    try {
-      RestResponse response = D2ClientUtils.sendD2GetRequest(requestPath, d2Client);
-      String responseBody = response.getEntity().asString(StandardCharsets.UTF_8);
-      return ObjectMapperFactory.getInstance().readValue(responseBody, responseClass);
-    } catch (Exception e) {
-      throw new VeniceException("Failed to get response for url: " + requestPath + " with D2 Client", e);
+    for (D2Client d2Client: d2ClientList) {
+      try {
+        RestResponse response = D2ClientUtils.sendD2GetRequest(requestPath, d2Client);
+        String responseBody = response.getEntity().asString(StandardCharsets.UTF_8);
+        return ObjectMapperFactory.getInstance().readValue(responseBody, responseClass);
+      } catch (Exception e) {
+        LOGGER.error("Failed to get response for url: " + requestPath + " with D2 Client:" + d2Client, e);
+      }
     }
+    throw new VeniceException("List of D2 Clients failed to get response for url: " + requestPath);
```

Comment:
Let's change the error message to the previous one. We don't need to call out if it was a List of D2 clients or a single D2 client